src_c/rect.c

```diff
@@ -1131,6 +1132,23 @@ pg_rect_length(PyObject *_self)
     return 4;
 }
 
+static int
+pg_rect_contains_multi(pgRectObject *argrect, PyObject *self_uncast) {
+    int contained;
+    GAME_Rect *self, temp;
+    
+    if (!(self = pgRect_FromObject(self_uncast, &temp))) {
+        return -1;
+    }
+
+    contained = (self->x <= argrect->r.x) && (self->y <= argrect->r.y) &&
+                (self->x + self->w >= argrect->r.x + argrect->r.w) &&
+                (self->y + self->h >= argrect->r.y + argrect->r.h) &&
+                (self->x + self->w > argrect->r.x) &&
+                (self->y + self->h > argrect->r.y);
+    return contained;
```

Comment:
seems like there's some code repetition here. The function `pg_rect_contains` above does almost exactly the same thing, except it returns a python bool object. Perhaps this function can be renamed to something like `_pg_rect_contains`, and then `pg_rect_contains` can call this function, something like 
```c
static PyObject *
pg_rect_contains(pgRectObject *self, PyObject *args)
{
    return PyBool_FromLong(_pg_rect_contains(self, args);
}
```

Reply:
Makes sense. Will go ahead and make it more modular!

Reply:
The other issue stems from the fact that __contains__ passes the argument as the first function. I'm still a little unskilled with using c and python together so if theres a way to typecast this properly to be able to easily flip arguments I will do this. Leaving as is for now!

Reply:
All done, added a reverse boolean to make sure that the reversal works properly!